src/main/kotlin/app/services/impl/HttpKeyService.kt

```diff
@@ -28,17 +28,27 @@ class HttpKeyService(private val httpClientProvider: HttpClientProvider) : KeySe
 
     companion object {
         val logger: Logger = LoggerFactory.getLogger(HttpKeyService::class.toString())
-        const val maxAttempts = 10
-        const val backoffMillis = 1000L
+
+        // Will retry at 1s, 2s, 4s, 8s, 16s then give up (after a total of 31 secs)
+        const val maxAttempts = 5
+        const val initialBackoffMillis = 1000L
+        const val backoffMultiplier = 2.0
     }
 
     @Override
-    @Retryable(value = [DataKeyServiceUnavailableException::class], maxAttempts = maxAttempts, backoff = Backoff(delay = backoffMillis))
+    @Retryable(value = [DataKeyServiceUnavailableException::class],
+        maxAttempts = maxAttempts,
+        backoff = Backoff(delay = initialBackoffMillis, multiplier = backoffMultiplier))
+    @Throws(DataKeyServiceUnavailableException::class)
     override fun batchDataKey(): DataKeyResult {
         try {
+            val dksUrl = "$dataKeyServiceUrl/datakey"
+            logger.info("dataKeyServiceUrl: '$dksUrl'.")
```

Comment:
log to `debug` maybe? might get a little spammy otherwise on retry attempts

Reply:
It's called fairly infrequently, so it's good to know when we are using it. I think.